spec/views/hq/users/index_spec.rb

```diff
@@ -2,51 +2,81 @@
+      end
+
+      it 'hides the table' do
+        expect(rendered).to match /No Users Found/
+        expect(rendered).to have_selector('table#users-index', count: 0)
+      end
+
+      it 'hides the pagination bar' do
+        expect(rendered).to_not have_selector('.pagination', text: /Previous(.*)1(.*)Next/)
       end
     end
 
     context '1 user' do
-      it 'displays a single user' do
-        @users = [ double('User', user_name: 'Frederick Bloggs', 
-          email: 'fred.bloggs@example.com', active_sponsors: (1..5).to_a)]
+      before(:each) do
+        @users = [ FactoryGirl.build_stubbed(:user, 
+                   user_name: 'Frederick Bloggs',
+                   email: 'fred.bloggs@example.com') ].paginate
         render
-        expect(rendered).to have_selector('table tr', count: 2)
-        expect(rendered).to have_link('Frederick Bloggs', href: hq_user_path(@users[0]))
+      end
+
+      it 'displays a single user' do
+        expect(rendered).to have_selector('table tbody tr', count: 1)
+        expect(rendered).to have_link('Frederick Bloggs', href: hq_user_path(@users.first))
         expect(rendered).to match 'fred.bloggs@example.com'
-        expect(rendered).to match '5'
+      end
+
+      it 'hides the pagination bar' do
+        expect(rendered).not_to have_selector('.pagination', text: /Previous(.*)1(.*)Next/)
       end
     end
 
-    context '>1 user' do
+    context 'many users on multiple pages' do
+      before :each do
+        @users= [ FactoryGirl.build_stubbed(:user),
+                  FactoryGirl.build_stubbed(:user, user_name: 'Joseph Soap',
+                    email: 'joe.soap@example.com'),
+                  FactoryGirl.build_stubbed(:user, user_name: 'Tom Jones',
+                    email: 'tom.jones@example.com') ].paginate(:page => 2, :per_page => 2)
+      end
+
       it 'displays multiple users' do
-        @users = [ double('User', user_name: 'Frederick Bloggs', 
-          email: 'fred.bloggs@example.com', active_sponsors: (1..5).to_a), 
-                   double('User', user_name: 'Joseph Soap', 
-          email: 'joe.soap@example.com', active_sponsors: (1..7).to_a) ]
         render
-        expect(rendered).to have_selector('table tr', count: 3)
-        expect(rendered).to have_link('Frederick Bloggs', href: hq_user_path(@users[0]))
-        expect(rendered).to match 'fred.bloggs@example.com'
-        expect(rendered).to match '5'
-        expect(rendered).to have_link('Joseph Soap', href: hq_user_path(@users[1]))
-        expect(rendered).to match 'joe.soap@example.com'
-        expect(rendered).to match '7'
+
+        expect(rendered).to have_selector('table tbody tr', count: 1)
+        expect(rendered).to have_link('Tom Jones', href: hq_user_path(@users.first))
+        expect(rendered).to match 'tom.jones@example.com'
+        expect(rendered).to_not have_link('Joseph Soap')
+        expect(rendered).to_not match 'joe.soap@example.com'
```

Comment:
Sorry, I still think that this is unnecessary and brittle. If the view renders a single user's info correctly (preceding test), there is no reason to think that it won't render multiples correctly as well. The only thing possibly worth checking is that we included the required fields, and the single-user test accomplishes that. Parsing a collection and displaying each record is handled by Rails and so is not subject to testing here.

Testing that the table body on page 2 only has one row _and_ that it only displays the last user _and_ that the page shows the pagination bar is also redundant, as the three go hand-in-hand.

I also mentioned the issue of brittleness earlier - if we change the sort order of users from id to name or email, the test may break. To make things worse, the first user is created without explicit name or email, so the test may break _sometimes_ depending on `Faker`'s output, which is always a pain to trace.

On a higher level, I believe that this super-tight coupling of tests to objects under test is unnecessarily restrictive. We essentially duplicate the view file in the spec file by iterating over each single field. This kind of brute-force testing - "Let's type the same thing in multiple places so that if we forget to include something in one file, it will be caught by the other" - just feels clumsy and redundant. By that logic, we should duplicate everything again in a feature, to be extra-extra sure. And then have to change 3 files when the client decides not to display user email on the index page.

Our initial aim was to only test logic in view specs, such as conditional displays of elements, and I would really like to stick to that. Which is why I'm making so much noise over this little test :)